src/BenchmarksApps/Kestrel/PlatformBenchmarks/BenchmarkApplication.HttpConnection.cs

```diff
@@ -140,51 +146,35 @@ private async Task ProcessRequestsAsync()
             }
         }
 #endif
-        private bool ParseHttpRequest(ref ReadOnlySequence<byte> buffer, bool isCompleted, out SequencePosition examined)
+        private bool ParseHttpRequest(ref SequenceReader<byte> reader, in ReadOnlySequence<byte> buffer, bool isCompleted)
         {
-            examined = buffer.End;
             var state = _state;
 
-            if (!buffer.IsEmpty)
+            if (state == State.StartLine)
             {
-                SequencePosition consumed;
-                if (state == State.StartLine)
+                var unconsumedSequence = buffer.Slice(reader.Position);
+                if (Parser.ParseRequestLine(new ParsingAdapter(this), unconsumedSequence, out var consumed, out _))
                 {
-                    if (Parser.ParseRequestLine(new ParsingAdapter(this), buffer, out consumed, out examined))
-                    {
-                        state = State.Headers;
-                    }
+                    state = State.Headers;
 
-                    buffer = buffer.Slice(consumed);
+                    var parsedLength = unconsumedSequence.Slice(reader.Position, consumed).Length;
+                    reader.Advance(parsedLength);
```

Comment:
This is a bit horrible; but can't advance/reposition a `SequenceReader` with a `SequencePosition`.

However want to change `Parser.ParseRequestLine` to take a `SequenceReader` so it's only short term

Reply:
Looks like there is an api proposal to fix this? https://github.com/dotnet/runtime/issues/30770

/cc @davidfowl